docs/reference/userapi.rst

```diff
@@ -70,3 +76,129 @@ that best suits that file-format.
+    # img = iio.v2.mvolread(image_resource)
+    img = [im for im in iio.v3.imiter(image_resource)]
+
+Writing Images
+--------------
+
+Similar to reading images, the new `iio.imwrite` can handle ndimages.
+``iio.mimwrite``, ``iio.volwrite``, and ``iio.mvolwrite`` have all dissapeared.
+The same goes for their aliases ``iio.mimsave``, ``iio.volsave``,
+``iio.mvolsave``, and ``iio.imsave``. They are now all covered by
+``iio.imwrite``.
+
+To reproduce similar behavior to the v2 API behavior use one of the following
+snippets::
+
+    # img = iio.v2.imwrite(image_resource, ndimage)
+    # img = iio.v2.volwrite(image_resource, ndimage)
+    img = iio.v3.imwrite(image_resource, ndimage)
+
+    # img = iio.v2.mimwrite(image_resource, list_of_ndimages)
+    # img = iio.v2.mvolwrite(image_resource, list_of_ndimages)
+    img = iio.v3.imwrite(image_resource, list_of_ndimages)
+
+Reader and Writer
+-----------------
+
+Plugins now longer have nested ``Reader`` and ``Writer`` classes and they are no
+longer global singletons. Instead, a new plugin object is instantiated in mode ``r``
+(reading) or  ``w`` (writing) for each `ImageResource` being opened.
+As such, the functions ``iio.get_reader``, ``iio.get_writer``, ``iio.read``, and
+``iio.save`` have become obsolete and the plugin object is used directly
+instead. Using these is considered advanced usage, happens inside a context
+manager, and uses `iio.imopen`::
+
+    # reader = iio.get_reader(image_resource)
+    # reader = iio.read(image_resource)
+    with iio.imopen(image_resource, "r") as reader:
+        # file is only open/accessible here
+        # img = reader.read(index=0)
+        # meta = reader.metadata(index=0)
+        # ...
+        pass
+
+
+    # writer = iio.get_writer(image_resource)
+    # writer = iio.save(image_resource)
+    with iio.imopen(image_resource, "w") as writer:
+        # file is only open/accessible here
+        # writer.write(ndimage)
+        # ...
+        pass
+
+Metadata
+--------
+
+The v3 API makes handling of metadata much more consistent and explicit.
+Previously in v2, metadata was provided as a ``dict`` that was attached to the
+image by returning a custom subclass of ``np.ndarray``. Now metadata is
+served independently from pixel data::
+
+    # metadata = iio.imread(image_resource).meta
+    metadata = iio.immeta(image_resource)
+
+Further, ImageIO now provides a curated set of standardized metadata, which is
+called :class:`ImageProperties <imageio.core.v3_plugin_api.ImageProperties>`, in
+addition to above metadata. The difference between the two is as follows: The
+metadata dict contains metadata using format-specific keys to the extent the
+reading plugin supports them. The ImageProperties dataclass contains generally
+available metadata using standardized attribute names. Each plugin provides the
+same set of properties, and if the plugin or format doesn't provide a field it
+is set to a (sensible) default value. To access ImageProperties use::
```

Comment:
or to None, still TBD

Reply:
True. My thinking here was that "sensible default value" admits to the interpretation of: "it is always None if not provided", so we we are correct in either case x)

Reply:
Ok, but would still be good to mention that values can be `None` so the user can take that into account. If we go that route.

Reply:
agreed. Does this mean we have a decision for https://github.com/imageio/imageio/issues/754? 

Either way, I have changed this around and added the phrase "; usually `None`". I think this is as close as we get until we make the definite decision to return `None` when the value is not set.

Reply:
Yes, I think it makes sense.

Reply:
Perfect. That was the last open point in this review. Any other points or shall we merge?